dart/tool/java/generate_java.dart

```diff
@@ -133,12 +133,12 @@ class Api extends Member with ApiParseUtil {
     for (var m in methods) {
       for (var a in m.args) {
         if (a.hasDocs) continue;
-        var t = types.firstWhere((t) => t.name == a.type, orElse: () => null);
```

Comment:
@devoncarew  The `==` for compatible type lint is no nice. Take a look at this. Clearly there is some behavior here that was not working

Reply:
Wow, good catch with this lint. Will run locally to see if I can determine the intent of this code.